cmd/kube-apiserver/app/options/options.go

```diff
@@ -134,14 +134,32 @@ func NewServerRunOptions() *ServerRunOptions {
 	return &s
 }
 
+func addDummyInsecureFlags(fs *pflag.FlagSet) {
+	var (
+		bindAddr = net.IPv4(127, 0, 0, 1)
+		bindPort int
+	)
+
+	for _, name := range []string{"insecure-bind-address", "address"} {
+		fs.IPVar(&bindAddr, name, bindAddr, ""+
+			"The IP address on which to serve the insecure port (set to 0.0.0.0 for all IPv4 interfaces and :: for all IPv6 interfaces).")
+		fs.MarkDeprecated(name, "This flag has no effect now and will be removed in a future version.")
+	}
+
+	for _, name := range []string{"insecure-port", "port"} {
+		fs.IntVar(&bindPort, name, bindPort, ""+
+			"The port on which to serve unsecured, unauthenticated access.")
+		fs.MarkDeprecated(name, "This flag has no effect now and will be removed in a future version.")
+	}
```

Comment:
the only other thing I think would be good is if we can fail if someone tries to explicitly set the port args to a non-zero value... the reason we're keeping these registered is for people who were explicitly disabling the ports

Reply:
The value of these flags is checked now.